src/selectors/protocol.js

```diff
@@ -2,7 +2,26 @@
 
 import { createDeepEqualSelector } from './utils';
 
+export const protocolRegistry = createDeepEqualSelector(
+  state => state.protocol.variableRegistry,
+  registry => registry,
+);
+
+export const protocolForms = createDeepEqualSelector(
+  state => state.protocol.forms,
+  forms => forms,
+);
+
 export const getExternalData = createDeepEqualSelector(
   state => state.protocol.externalData,
   protocolData => protocolData,
 );
+
+export const getNodeColor = createDeepEqualSelector(
```

Comment:
Any selectors using props need to be function factories to cache the results properly. Otherwise all instances of the components will share the same cache. e.g.

```js
export const makeGetNodeColor = () =>
  createDeepEqualSelector(
    protocolRegistry,
    (state, props) => props.nodeType,
    ({ node }, nodeType) => node && node[nodeType] && node[nodeType].color,
  );
```